aks/locals.tf

```diff
@@ -1,6 +1,16 @@
 locals {
   use_managed_identity = try(var.metrics_storage.managed_identity_node_rg_name != null, false)
 
+  metrics_storage = var.metrics_storage != null ? {
+    storage_config = merge({ type = "AZURE" }, {
```

Comment:
If I'm not mistaken, this `merge()` is useless.

Reply:
Good catch! I cannot test it, but the changes I made should be equivalent.